web/src/components/ErrorScreen.js

```diff
@@ -1,122 +1,138 @@
-              <p>This isn't normal at all, so I'm not really sure how to proceed...</p>
-              <p>Maybe try refreshing? Let's hope for the best!</p>
-            </section>
-          ),
-        };
-
+        return this.renderNoAuth();
       case 'not connected':
-        return {
-          header: 'I fell asleep!',
-          body: (
-            <section>
-              <p>Give me a couple of seconds to wake up, then try refreshing the page!</p>
-            </section>
-          ),
-        };
-
+        return this.renderNoConnection();
       case 'not in server':
-        return {
-          // TODO
-          header: 'Access denied!',
-          body: (
-            <div>
-              <section>
-                You need to be on:
-                <Guild />
-                {/* TODO */}
-              </section>
-              <section>
-                You are logged in as:
-                <div className="member">
-                  <UserAvatar user={data.user.id} avatar={data.user.avatar} />
-                  <div className="member-inner">
-                    <div className="member-username">
-                      <span className="member-username-inner">{data.user.username}</span>
-                    </div>
-                    <span className="member-discriminator">#{data.user.discriminator}</span>
-                  </div>
-                </div>
-              </section>
-              <section>
-                <p>
-                  <strong>If this is the correct account</strong>, you're out of luck, buddy.
-                </p>
-                <p>
-                  <strong>If this is the wrong account</strong>,{' '}
-                  <a href="https://discordapp.com/channels/@me">open Discord in your browser</a> and log into the
-                  correct account. Then click the button below.
-                </p>
-              </section>
-            </div>
-          ),
-          footer: (
-            <form method="POST" action="logout">
-              <button className="primary" type="submit">
-                Re-authenticate
-              </button>
-            </form>
-          ),
-        };
-
+        return this.renderNoAccess(data);
       case 'disconnected':
-        return {
-          header: 'Disconnected!',
-          body: (
-            <section>
-              <p>Sorry! I might have gone to sleep for the night!</p>
-              <p>If you know I'm supposed to be awake, try giving this page a refresh!</p>
-            </section>
-          ),
-        };
-
+        return this.renderDisconnected();
       default:
-        return {
-          // TODO
-          header: 'Wha?!',
-          body: (
-            <section>
-              <p>???</p>
-            </section>
-          ),
-        };
+        return null;
     }
   }
+  render() {
+    return (
+      <CSSTransitionGroup
+        transitionName="fadeError"

```

Comment:
this is a generic classname that gets applied for transitions

this *may not always be a fade* so calling it "`fadeError`" may become misleading in the future

Reply:
🐇